charts/federation-v2/charts/controllermanager/templates/federationconfig.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederationConfig
+metadata:
+  name: federation-v2
+  namespace: {{ .Release.Namespace }}
+spec:
+  limited-scope: {{ .Values.global.limitedScope }}
+{{- if .Values.registryNamespace }}
+  registry-namespace: {{ .Values.registryNamespace }}
+{{- else }}
+{{- if .Values.global.limitedScope }}
+  registry-namespace: {{ .Release.Namespace }}
+{{- else }}
+  registry-namespace: kube-multicluster-public
+{{- end }}
+{{- end }}
+  controller-duration:
+    available-delay: 20s
+    unavailable-delay: 60s
+    cluster-monitor-period: 40s
+  leader-elect:
+    lease-duration: 15s
+    renew-deadline: 10s
+    retry-period: 5s
+    resource-lock: configmaps
```

Comment:
In the current format, it looks like the above parameters are hardcoded and there is no option to user to configure them according to their environment. Please expose them in Values.yaml of the chart.

Also, not specific to this pr. we should introduce defaulting the fields in API types, if they are not specified by user (or by helm).

Reply:
Defaulting support for CRDs is proposed but the PR appears to have been abandoned: https://github.com/kubernetes/kubernetes/pull/63604

That means that defaulting will require webhooks (yuck).

Reply:
I think we can use condition check in helm chart currently to make it work.
If related value is defined, we use it. Or else, use hard coded default value.